cis194/week2/tien/homework2/src/LogAnalysis.hs

```diff
@@ -0,0 +1,47 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module LogAnalysis where
+
+import           Log
+import           Text.Read (readMaybe)
+
+getHead :: [String] -> String
+getHead wordList =
+  case wordList of
+    x:_ -> x
+    _   -> ""
+
+getNthValue :: Int -> String -> Maybe Int
+getNthValue n str = readMaybe (getHead . drop (n-1) . words $ str) :: Maybe Int
+
+getNthTailConent :: Int -> String -> String
+getNthTailConent n str = unwords . drop n . words $ str
+
+parseMaybeInt :: Maybe Int -> Int
+parseMaybeInt maybeValue =
+  case maybeValue of
+    Just i  -> i
+    Nothing -> 0
+
+parseMessage :: String -> LogMessage
+parseMessage logStr =
+  case getHead . words $ logStr of
+    "I" ->
+      LogMessage
+        Info
+        (parseMaybeInt . getNthValue 2 $ logStr)
```

Comment:
When `parseMaybeInt`gives 0, it should be a `Unknown` message. Try more pattern matches first to get the correct behaviour.